crates/steel-core/src/parser/expander.rs

```diff
@@ -625,251 +662,235 @@ impl MacroPattern {
     //     }
     // }
 
-    // TODO: Don't do this?
-    pub fn deconstruct_without_syntax(&self) -> SmallVec<[&InternedString; 8]> {
-        match self {
-            Self::Single(s) => smallvec![s],
-            Self::Many(s) => smallvec![s],
-            Self::Nested(v) => v
-                .iter()
-                .flat_map(|x| x.deconstruct_without_syntax())
-                .collect(),
-            Self::ManyNested(v) => v
-                .iter()
-                .flat_map(|x| x.deconstruct_without_syntax())
-                .collect(),
-            _ => smallvec![],
-        }
-    }
-
     pub fn is_many(&self) -> bool {
         matches!(self, MacroPattern::Many(_) | MacroPattern::ManyNested(_))
     }
 }
 
-pub fn match_vec_pattern(args: &[MacroPattern], list: &[ExprKind]) -> bool {
-    let mut token_iter = list.iter();
-
-    for pat in args {
-        if let Some(val) = token_iter.next() {
-            // dbg!(&pat);
-            // println!("{}", val);
-
-            match pat {
```

Comment:
:information_source: This diff looks big, but this `match` is moved almost verbatim to line 727. This helper is much flatter now.